File: beecrowd/python/estruturas-e-bibliotecas/ex1068.py
def validar_parenteses(entrada):
    pilha = []
    for caractere in entrada:
        if caractere == '(':
            pilha.append('(')
        elif caractere == ')':
            if len(pilha) > 0:
                pilha.pop()
            else:
                pilha.append(')')
                break
    return True if len(pilha) == 0 else False

while True:
    try:
        entrada = input()
        print("correct") if validar_parenteses(entrada) else print("incorrect")
    except EOFError:
        break

File: beecrowd/c++/estruturas-e-bibliotecas/ex1068.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

bool isValid(string expressao){
    stack<char> pilha;
    for(char c : expressao){
        if (c=='(') pilha.push('(');
        else if (c==')'){
            if (pilha.empty()) return false;
            else pilha.pop();
        }
    }

    if (pilha.empty()) return true;
    else return false;
}

int main(){
    string expressao;
    while(getline(cin, expressao)){
        if(isValid(expressao)) cout<<"correct"<<endl;
        else cout<<"incorrect"<<endl;
    }
    return 0;
}